storage/pebble/registers.go

```diff
@@ -171,3 +183,22 @@ func convertNotFoundError(err error) error {
 	}
 	return err
 }
+
+// updateBounds updates the stored height bounds based on the latest block height.
+// This method also handles marks older data as pruned if the prune threshold is exceeded.
+func (s *Registers) updateBounds(latestHeight uint64) {
```

Comment:
We only need to update the latest height, no need to keep track of the first height. 

The first height is supposed to be calculated by `s.latestHeight.Load() - s.pruneThreshold`, and the calculation should happen in `FirstHeight()` method

Reply:
there are a few corner cases to handle when calculating the first height. I suggested we should just track it on the storage object since writes are much less frequent than reads.

The idea is that anything below `s.latestHeight.Load() - s.pruneThreshold` should be assumed to be pruned from the db. so maintaining a pointer to lower values doesn't seem useful. it will also require that we have a callback to update it once pruning is implemented.

Reply:
@Guitarheroua Leo and I synced on this. He convinced me switching to calculating the first height will reduce the complexity here.

the ask is to move the calculation logic into `FirstHeight()`, and drop the `firstHeight` atomic